prowler/providers/aws/services/cognito/cognito_user_pool_self_registration_disabled/cognito_user_pool_self_registration_disabled.py

```diff
@@ -0,0 +1,47 @@
+from prowler.lib.check.models import Check, Check_Report_AWS
+from prowler.providers.aws.services.cognito.cognito_identity_client import (
+    cognito_identity_client,
+)
+from prowler.providers.aws.services.cognito.cognito_idp_client import cognito_idp_client
+
+
+class cognito_user_pool_self_registration_disabled(Check):
+    def execute(self):
+        findings = []
+        for user_pool in cognito_idp_client.user_pools.values():
+            report = Check_Report_AWS(self.metadata())
+            report.region = user_pool.region
+            report.resource_id = user_pool.id
+            report.resource_arn = user_pool.arn
+            report.resource_tags = user_pool.tags
+            report.status = "PASS"
+            report.status_extended = (
+                f"User pool {user_pool.id} has self registration disabled."
+            )
+            if not user_pool.admin_create_user_config.get(
```

Comment:
Could you store this into the class object instead of doing a get in the map?

Reply:
Done